ocaml/xapi/xapi_host_helpers.ml

```diff
@@ -286,6 +286,21 @@ module InitiatorName = struct
     let config_file = make_initiatorname_config iqn hostname in
     Unixext.write_string_to_file !Xapi_globs.iscsi_initiator_config_file config_file
 
+  let set_multipathing enabled =
+    let flag = !Xapi_globs.multipathing_config_file in
+    if enabled then Unixext.touch_file flag
+    else begin
+      if Sys.file_exists flag then Sys.remove flag
+    end
+
+  let sync_config_files ~__context =
+    (* If the host fields are not in sync with the values in other_config,
+       the other_config watcher thread will make sure that these functions will
+       be called again with the up to date values. *)
+    let self = Helpers.get_localhost ~__context in
+    set_initiator_name (Db.Host.get_iscsi_iqn ~__context ~self);
```

Comment:
Can this fail when there is no iscsi IQN configured?

Reply:
Right now will not fail, as it's a string, and it's the empty string by default. It will just write the empty string to the config file.

Reply:
It's empty by default in the datamodel though 🤔

Reply:
Yes empty = don't change it, I think.

Reply:
Nevermind, looks like this is set by firstboot scripts, so lets not worry about empty IQN.

Reply:
Sounds good. As discussed, I'll disallow empty IQNs in host.set_iscsi_iqn